### PR TITLE
add "homectl firstboot" verb, that runs at first boot and can create a user, interactively or from creds

### PR DESCRIPTION
**Pull request by poettering**
This is kinda the continuation of systemd-firstboot and creates a user ineractively if no regular user exists. Or alternatively it creates a user based on passed credentials. Very similar to systemd-firstboot

### CODE REVIEW

**Comment by DaanDeMeyer** [comment]
Maybe set this up by default in the mkosi image using credentials?

**Comment by poettering** [comment]
> I was hoping we'd use the credentials instead of turning off firstboot ...

Well, the problem is to create a user with homed we must somehow also supply the raw password because we need that for the LUKS setup. So for now I figured we just turn this off, so that the tests pass, and then I'll add another set of creds that allow passing in the password raw. and then we can set up things via mkosi so that a useful user is created by default.

Or in other words: this is the first step only.

File: src/home/homectl.c [hunk 1]
```diff
@@ -2142,6 +2157,190 @@ static int rebalance(int argc, char *argv[], void *userdata) {
...
+                username = mfree(username);
+
+                r = ask_string(&username,
+                               "%s Please enter user name to create (empty to skip): ",
+                               special_glyph(SPECIAL_GLYPH_TRIANGULAR_BULLET));
+                if (r < 0)
+                        return log_error_errno(r, "Failed to query user for username: %m");
+
+                if (isempty(username)) {
+                        log_info("Not data entered, skipping.");
```
Comment (daandemeyer) [inline src/home/homectl.c]: "No data entered, skipping"

File: mkosi.conf [hunk 2]
```diff
@@ -37,3 +35,4 @@ KernelCommandLineExtra=systemd.crash_shell
                        selinux=0
                        enforcing=0
                        systemd.early_core_pattern=/core
+                       systemd.firstboot=yes
```
Comment (daandemeyer) [inline mkosi.conf]: Shouldn't this be `no`?

**Review by DaanDeMeyer — APPROVED** [review]

**Comment by poettering** [comment]
forced pushed new version, ptal!

**Review by DaanDeMeyer — APPROVED** [review]

**Pull request merged**